src/include/sof/audio/sink_api.h

```diff
@@ -0,0 +1,163 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ */
+
+#ifndef __SOF_SINK_API_H__
+#define __SOF_SINK_API_H__
+
+#include <sof/common.h>
+#include <errno.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * this is a definition of API to sink of audio data
+ *
+ * THE SINK is any component that can store data somehow and provide a buffer to be filled
+ * with data at request. The sink API does not define how the data will be processed/used
+ *
+ * The user - a module - sees this API as a destination it must send data to
+ * The IMPLEMENTATION - audio_stream, DP Queue - sees this as a producer that
+ *			PROVIDES data for processing
+ *
+ * Examples of components that should expose SINK api
+ * - /dev/null
+ *     all the data stored in sink buffer are just simply discarded
+ * - I2S sender
+ *     Data stored in sink buffer will be sent to the external world
+ * - a memory ring buffer
+ *     data stored in the buffer will be sent to another module (usually using source API, but it
+ *     does not matter in fact).
+ *
+ *  The main advantage of using sink API instead of just taking pointers to the buffers is that
+ *  the buffer may be prepared at the moment the data producer is requesting it. i.e.
```

Comment:
'may be prepared' is ambiguous, it could mean

a) may be in prepared state ('prepared' is an adjective in your sentence)
b) may be transitioned to prepared state ('prepared is a verb in your sentence)

Which is it?